forms-flow-api/src/formsflow_api/services/draft.py

```diff
@@ -159,3 +159,22 @@ def make_submission_from_draft(data: Dict, draft_id: str, token=None, **kwargs):
         )
         ApplicationService.start_task(mapper, payload, token, application)
         return application
+
+    @staticmethod
+    @user_context
+    def delete_draft(draft_id: int, **kwargs):
+        """Delete draft."""
+        user: UserContext = kwargs["user"]
+        user_id: str = user.user_name
+        draft = Draft.get_by_id(draft_id=draft_id, user_id=user_id)
+        if draft:
+            application = Application.find_by_id(application_id=draft.application_id)
+            # deletes the draft and application entry related to the draft.
+            draft.delete()
```

Comment:
doesn't seem like the right way to do it. @auslin-aot can u check the `cascade` option, in models? When we delete an entry the associated entries also need to be deleted by default. @sumesh-aot thoughts?

Reply:
Added cascade option.